### PR TITLE
[OpenGL] Add Support for DotNet Attributes

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: src/OpenGL/CGLContext.cs [hunk 1]
```diff
@@ -30,9 +30,17 @@
...
 namespace OpenGL {
-	[Deprecated (PlatformName.MacOSX, 10, 14, message : "Use 'Metal' Framework instead.")]
+	#if !NET
+			[Deprecated (PlatformName.MacOSX, 10, 14, message: "Use 'Metal' Framework instead.")]
+	#else
+			[UnsupportedOSPlatform ("macos10.14")]
+	#if MONOMAC
+			[Obsolete ("Starting with macos10.14 Use 'Metal' Framework instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+	#endif
+	#endif
```
Comment (spouliot) [inline src/OpenGL/CGLContext.cs:43]: for future PR do not indent the `#if`
that will make the `[Deprecated ...]` shows as not changed (by default) and make PR easier to review

Comment (spouliot) [reply src/OpenGL/CGLContext.cs:43]: unless this is how @mandel-macaque macros/snippets are working ?

**Review by spouliot — APPROVED** [review]

File: src/OpenGL/CGLContext.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/OpenGL/CGLContext.cs:43]: That was manual, the snippets don't do that

**Review by chamons — APPROVED** [review]

File: src/OpenGL/CGLContext.cs (hunk 1 above)
Comment (tj-devel709) [reply src/OpenGL/CGLContext.cs]: Sorry, don't know why I did that haha

**Review by spouliot — APPROVED** [review]

**Pull request merged**